File: Python3/0253-Meeting-Rooms-II/soln-4.py
class Solution:
    def minMeetingRooms(self, intervals: List[List[int]]) -> int:
        times = collections.Counter()
        for s, e in intervals:
            times[s] += 1
            times[e] -= 1
        mx = cur = 0
        for t in sorted(times):
            cur = cur + times[t]
            mx = max(mx, cur)
        return mx

File: C++/0253-Meeting-Rooms-II/soln-4.cpp
class Solution {
public:
    int minMeetingRooms(vector<vector<int>>& intervals) {
        map<int, int> times;
        for(const auto & interval : intervals) {
            ++times[interval[0]];
            --times[interval[1]];
        }
        int cnt = 0;
        int mx = 0;
        for(auto & p : times) {
            cnt += p.second;
            mx = max(mx, cnt);
        }
        return mx;
    }
};
